### PR TITLE
[LM-11579] Adding LMDocument

### PR DESCRIPTION
**Pull request by divyanshgandhilm**
## Description
Adding PDF documents attachment support

**Features**
- Added LMDocument Widget
- Updated media_utils and screens
- Cleaned up code

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [ ] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [ ] Label added
- [ ] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [ ] Description about the Feature/ Bugs added

### CODE REVIEW

File: lib/packages/ui/lib/src/widgets/media/document/preview.dart [hunk 1]
```diff
@@ -0,0 +1,184 @@
+import 'package:flutter/material.dart';
+import 'package:likeminds_chat_flutter_ui/likeminds_chat_flutter_ui.dart';
+import 'package:path/path.dart';
+
+///{@template lm_chat_document_preview}
+/// A widget that displays a document preview with multiple attachments.
```
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/media/document/preview.dart:6]:
```suggestion
/// A widget that displays a document preview with single attachment.
```

File: lib/packages/ui/lib/src/widgets/media/document/preview.dart
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/media/document/preview.dart:1]: how will we add customisation here?

File: lib/packages/ui/lib/src/widgets/media/document/thumbnail.dart [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+  Widget build(BuildContext context) {
+    return FutureBuilder(
+      future: loadedFile,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.done &&
+            snapshot.hasData) {
+          return InkWell(
+            onTap: () async {
+              if (widget.media.mediaUrl != null) {
+                debugPrint(widget.media.mediaUrl);
```
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/media/document/thumbnail.dart:74]: remove debugPrint

File: lib/packages/ui/lib/src/widgets/media/document/thumbnail.dart [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                Uri fileUrl = Uri.parse(widget.media.mediaUrl!);
+                launchUrl(fileUrl, mode: LaunchMode.externalApplication);
+              } else {
+                // OpenFilex.open(file!.path);
+                debugPrint(file!.path);
```
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/media/document/thumbnail.dart:79]: remove debugPrint() and commented code.

File: lib/packages/ui/lib/src/widgets/media/document/tile.dart [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+///{@template lm_chat_document_tile}
+/// A widget that displays a tile for a document.
+///{@endtemplate}
+class LMChatDocumentTile extends StatefulWidget {
+  /// The document to display.
+  final LMChatMediaModel media;
+
+  /// The title of the document.
+  final Widget? title;
+
```
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/media/document/tile.dart:18]: add titleBuilder

File: lib/packages/ui/lib/src/widgets/media/document/tile.dart [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+  /// The subtitle of the document.
+  final Widget? subtitle;
```
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/media/document/tile.dart:20]: add subTitleBuilder

File: lib/packages/ui/lib/src/widgets/media/document/tile.dart [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+  /// The icon to display for the document.
+  final LMChatIcon? documentIcon;
```
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/media/document/tile.dart:23]: add iconBuilder

File: lib/packages/ui/lib/src/widgets/shimmers/document_shimmer.dart [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    return Container(
+      height: 78,
+      width: 60.w,
+      margin: const EdgeInsets.only(bottom: 10),
+      decoration: BoxDecoration(
+          border: Border.all(color: LMChatDefaultTheme.greyColor, width: 1),
+          borderRadius: BorderRadius.circular(kBorderRadiusMedium)),
+      padding: const EdgeInsets.all(kPaddingLarge),
+      child: Shimmer.fromColors(
+        baseColor: Colors.black26,
```
Comment (AnujLM) [inline lib/packages/ui/lib/src/widgets/shimmers/document_shimmer.dart:23]: make it customisable, by creating a style class and use here.

**Review by AnujLM — CHANGES_REQUESTED** [review]

File: lib/packages/ui/lib/src/widgets/media/document/preview.dart
Comment (divyanshgandhilm) [reply lib/packages/ui/lib/src/widgets/media/document/preview.dart:1]: Added style classes for both preview widgets

File: lib/packages/ui/lib/src/widgets/media/document/tile.dart (hunk 6 above)
Comment (divyanshgandhilm) [reply lib/packages/ui/lib/src/widgets/media/document/tile.dart:23]: Converted it to widget class

**Review by AnujLM — APPROVED** [review]

**Pull request merged**